src/test/kotlin/com/terraformation/backend/seedbank/db/accessionStore/AccessionStorePermissionTest.kt

```diff
@@ -0,0 +1,117 @@
+package com.terraformation.backend.seedbank.db.accessionStore
+
+import com.terraformation.backend.db.AccessionNotFoundException
+import com.terraformation.backend.db.FacilityNotFoundException
+import com.terraformation.backend.db.OrganizationNotFoundException
+import com.terraformation.backend.seedbank.model.AccessionModel
+import io.mockk.every
+import java.time.LocalDate
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.security.access.AccessDeniedException
+
+internal class AccessionStorePermissionTest : AccessionStoreTest() {
+  @Test
+  fun `fetchOneById throws exception if user does not have permission`() {
+    val initial = store.create(AccessionModel(facilityId = facilityId))
+    assertNotNull(initial, "Should have created accession successfully")
+
+    every { user.canReadAccession(any()) } returns false
+
+    assertThrows<AccessionNotFoundException> { store.fetchOneById(initial.id!!) }
+  }
+
+  @Test
+  fun `update does not write to database if user does not have permission`() {
+    every { user.canUpdateAccession(any()) } returns false
+    val initial = store.create(AccessionModel(facilityId = facilityId))
+
+    assertThrows<AccessDeniedException> { store.update(initial.copy(numberOfTrees = 1)) }
+
+    val afterUpdate = store.fetchOneById(initial.id!!)
+    assertNotNull(afterUpdate, "Should be able to read accession after updating")
+    assertNull(afterUpdate.numberOfTrees, "Update should not have been written")
+  }
+
+  @Test
+  fun `fetchHistory throws exception if user does not have permission`() {
+    val initial = store.create(AccessionModel(facilityId = facilityId))
+
+    every { user.canReadAccession(any()) } returns false
+
+    assertThrows<AccessionNotFoundException> { store.fetchHistory(initial.id!!) }
+  }
+
+  @Test
+  fun `delete throws exception if user does not have permission`() {
+    every { user.canDeleteAccession(any()) } returns false
+    val initial = store.create(AccessionModel(facilityId = facilityId))
+
+    assertThrows<AccessDeniedException> { store.delete(initial.id!!) }
+  }
+
+  @Test
+  fun `update ignores received and collected date edits for accessions from web`() {
```

Comment:
This one isn't permission-related.

Reply:
Not sure where is an obvious place for it. Maybe `AccessionDatabaseTest` since it is testing the database behavior upon receiving a web-originated accession? What do you think?

Reply:
Sure, that seems reasonable.